From c07238ae057f362a1ee301db6abb0b03fe822586 Mon Sep 17 00:00:00 2001
From: blackcathacker <douglas.pace@gmail.com>
Date: Thu, 7 May 2015 14:35:41 -0700
Subject: [PATCH] RESKC-393:Ensure GlobalVariables is updated in IncidentReport

The incident report in some cases(clicking cancel button for instance) doesn't update the GlobalVariables user session. This checks for and logs incidents of this nature but also overwrites the GlobalVariables with the request user session to ensure it is up to date.
---
 .../controller/interceptor/KcUifHandlerExceptionResolver.java | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/coeus-impl/src/main/java/org/kuali/coeus/sys/framework/controller/interceptor/KcUifHandlerExceptionResolver.java b/coeus-impl/src/main/java/org/kuali/coeus/sys/framework/controller/interceptor/KcUifHandlerExceptionResolver.java
index bba7a10..2f1a5e5 100644
--- a/coeus-impl/src/main/java/org/kuali/coeus/sys/framework/controller/interceptor/KcUifHandlerExceptionResolver.java
+++ b/coeus-impl/src/main/java/org/kuali/coeus/sys/framework/controller/interceptor/KcUifHandlerExceptionResolver.java
@@ -45,7 +45,8 @@
  */
 //not exactly sure how to do this with autowiring since exception handlers are special
 public class KcUifHandlerExceptionResolver implements org.springframework.web.servlet.HandlerExceptionResolver, Ordered {
-    HandlerExceptionResolver innerHandler;
+	private static final Logger LOG = Logger.getLogger(KcUifHandlerExceptionResolver.class);
+	HandlerExceptionResolver innerHandler;
 
     public int getOrder() {
         return Integer.MIN_VALUE;
@@ -53,6 +54,12 @@ public int getOrder() {
 
     @Override
     public ModelAndView resolveException(HttpServletRequest request, HttpServletResponse response, Object handler, Exception ex) {
+    	UserSession requestSession = (UserSession) request.getSession().getAttribute(KRADConstants.USER_SESSION_KEY);
+		if (System.identityHashCode(GlobalVariables.getUserSession()) != System.identityHashCode(requestSession)) {
+    		LOG.error("User Session Difference Detected. GlobalVariables session = " + GlobalVariables.getUserSession() + ", request session = " + requestSession);
+    	}
+    	GlobalVariables.setUserSession(requestSession);
+    	
         //Avoids NPE in rice incident handler
         if (handler == null) {
             return innerHandler.resolveException(request, response, NullHandler.INSTANCE, ex);
@@ -60,7 +67,7 @@ public ModelAndView resolveException(HttpServletRequest request, HttpServletResp
 
         return innerHandler.resolveException(request, response, handler, ex);
     }
-
+    
     public void setInnerHandler(HandlerExceptionResolver innerHandler) {
         this.innerHandler = innerHandler;
     }
